src/pages/Earn/EarnV2.tsx

```diff
@@ -0,0 +1,479 @@
+    gmxAddress,
+    esGmxAddress,
+    stakedGmxTrackerAddress,
+    extendedGmxTrackerAddress,
+    bnGmxAddress,
+    glpAddress,
+  ];
+  const rewardTrackersForDepositBalances = [
+    stakedGmxTrackerAddress,
+    stakedGmxTrackerAddress,
+    bonusGmxTrackerAddress,
+    feeGmxTrackerAddress,
+    feeGmxTrackerAddress,
+    feeGlpTrackerAddress,
+  ];
+  const rewardTrackersForStakingInfo = [
+    stakedGmxTrackerAddress,
+    bonusGmxTrackerAddress,
+    feeGmxTrackerAddress,
+    stakedGlpTrackerAddress,
+    feeGlpTrackerAddress,
+    extendedGmxTrackerAddress,
+  ];
+
+  const { marketTokensData } = useMarketTokensData(chainId, { isDeposit: false });
+  const {
+    marketsTokensApyData,
+    marketsTokensIncentiveAprData,
+    glvTokensIncentiveAprData,
+    marketsTokensLidoAprData,
+    glvApyInfoData,
+  } = useGmMarketsApy(chainId);
+  const vestingData = useVestingData(account);
+
+  const { data: walletBalances } = useSWR(
+    [
+      `StakeV2:walletBalances:${active}`,
+      chainId,
+      readerAddress,
+      "getTokenBalancesWithSupplies",
+      account || PLACEHOLDER_ACCOUNT,
+    ],
+    {
+      fetcher: contractFetcher(signer, ReaderV2, [walletTokens]),
+    }
+  );
+
+  const { data: depositBalances } = useSWR(
+    [
+      `StakeV2:depositBalances:${active}`,
+      chainId,
+      rewardReaderAddress,
+      "getDepositBalances",
+      account || PLACEHOLDER_ACCOUNT,
+    ],
+    {
+      fetcher: contractFetcher(signer, RewardReader, [depositTokens, rewardTrackersForDepositBalances]),
+    }
+  );
+
+  const { data: stakingInfo } = useSWR(
+    [`StakeV2:stakingInfo:${active}`, chainId, rewardReaderAddress, "getStakingInfo", account || PLACEHOLDER_ACCOUNT],
+    {
+      fetcher: contractFetcher(signer, RewardReader, [rewardTrackersForStakingInfo]),
+    }
+  );
+
+  const { data: stakedGmxSupply } = useSWR(
+    [`StakeV2:stakedGmxSupply:${active}`, chainId, gmxAddress, "balanceOf", stakedGmxTrackerAddress],
+    {
+      fetcher: contractFetcher(signer, Token),
+    }
+  );
+
+  const { data: aums } = useSWR([`StakeV2:getAums:${active}`, chainId, glpManagerAddress, "getAums"], {
+    fetcher: contractFetcher(signer, GlpManager),
+  });
+
+  const { data: nativeTokenPrice } = useSWR(
+    [`StakeV2:nativeTokenPrice:${active}`, chainId, vaultAddress, "getMinPrice", nativeTokenAddress],
+    {
+      fetcher: contractFetcher(signer, Vault),
+    }
+  );
+
+  const { data: sbfGmxBalance } = useSWR(
+    [`StakeV2:sbfGmxBalance:${active}`, chainId, feeGmxTrackerAddress, "balanceOf", account ?? PLACEHOLDER_ACCOUNT],
+    {
+      fetcher: contractFetcher(undefined, Token),
+    }
+  );
+
+  const { gmxPrice } = useGmxPrice(chainId, { arbitrum: chainId === ARBITRUM ? signer : undefined }, active);
+
+  const gmxSupplyUrl = getServerUrl(chainId, "/gmx_supply");
+  const { data: gmxSupply } = useSWR([gmxSupplyUrl], {
+    fetcher: (args) => fetch(...args).then((res) => res.text()),
+  });
+
+  let aum;
```

Comment:
Will it work if you just calculate this once say in `onSuccess` of swr? I don't think that it's good place for calculations, as we don't use `aums`, but need just `aum`

Reply:
done